src/IdentityModel.OidcClient/Infrastructure/LogSerializer.cs

```diff
@@ -27,7 +29,7 @@ static LogSerializer()
         /// <returns></returns>
         public static string Serialize(object logObject)
         {
-            return JsonConvert.SerializeObject(logObject, jsonSettings);
+            return Enabled ? JsonConvert.SerializeObject(logObject, jsonSettings) : "";
```

Comment:
Can you change that to "Logging has been disabled" - and put a comment on the Enabled property to describe why it is there.